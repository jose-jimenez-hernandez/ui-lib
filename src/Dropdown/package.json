{
  "name": "@settle-ui/dropdown",
  "version": "0.0.0",
  "description": "dropdown description",
  "main": "../lib/dropdown.es.js",
  "module": "../lib/dropdown.es.js",
  "scripts": {
    "build": "vite build",
    "build:watch": "vite build --watch",
    "lint": "eslint ./src",
    "tsc": "tsc"
  },
  "types": "./src/dropdown.d.ts",
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
