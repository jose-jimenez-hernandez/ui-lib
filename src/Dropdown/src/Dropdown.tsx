import { Fragment } from 'react';
import { Menu, Transition } from '@headlessui/react';
import { ChevronDownIcon, DotsVerticalIcon } from '@heroicons/react/solid';
import { DropdownProps } from './DropdownType';
import { useStyles } from './styles';

const Dropdown = ({ className, label, disabled, children }: Partial<DropdownProps>) => {
  const styles = useStyles({
    className,
    disabled,
  });

  return (
    <Menu as="div" className={styles.container}>
      <div>
        <Menu.Button className={styles.buttonContainer} disabled={disabled}>
          <span className={label ? '' : 'sr-only'}>{label || 'Options'}</span>
          {label && <ChevronDownIcon className="-mr-1 ml-2 h-5 w-5" aria-hidden="true" />}
          {!label && <DotsVerticalIcon className="h-5 w-5" aria-hidden="true" />}
        </Menu.Button>
      </div>

      <Transition
        as={Fragment}
        enter="transition ease-out duration-100"
        enterFrom="transform opacity-0 scale-95"
        enterTo="transform opacity-100 scale-100"
        leave="transition ease-in duration-75"
        leaveFrom="transform opacity-100 scale-100"
        leaveTo="transform opacity-0 scale-95"
      >
        <Menu.Items className={styles.items}>{children}</Menu.Items>
      </Transition>
    </Menu>
  );
};

export default Dropdown;
