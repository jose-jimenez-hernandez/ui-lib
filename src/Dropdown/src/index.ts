export * from './DropdownType';
export { default as Dropdown } from './Dropdown';
