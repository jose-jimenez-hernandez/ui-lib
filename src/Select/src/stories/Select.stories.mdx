import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Select |

Select component to select value from options.

## When To Use

- A dropdown menu for displaying choices - an elegant alternative to the native `<select>` element.


## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 
| disabled | Whether the select menu is disabled | boolean | - |
| placeholder | Placeholder of select | ReactNode | - |
| status          | Status of input                                | 'success' , 'error',  'none'           | 'none'    |
| statusMessage   | Setatus messages                               | string                                 | '' |  
| hintMessage     | Exta information about the field               | strin                                  | ''        |
| label           | Field primary label name                       | string                                 | ''        |
| labelHint       | Field secondary label name                     | string                                 | ''        |

It accepts all props which native support.
