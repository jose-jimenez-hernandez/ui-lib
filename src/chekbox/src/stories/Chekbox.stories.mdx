import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Chekbox |

To trigger an operation.

## When To Use



## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 

It accepts all props which native support.
