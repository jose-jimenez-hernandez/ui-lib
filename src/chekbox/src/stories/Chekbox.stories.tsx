import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import { Story } from '@storybook/react';
import Chekbox from '../Chekbox';
import ChekboxDoc from './Chekbox.stories.mdx';

export default {
  title: 'Components/Chekbox',
  component: Chekbox,
  parameters: {
    docs: {
      page: (ChekboxDoc as any).parameters?.docs.page,
    },
  },
};

const ChekboxTemplate: Story = (args: any) => ( <Chekbox  {...args}/>
);

export const DefaultChekbox = ChekboxTemplate.bind({});
DefaultChekbox.args = {  };
