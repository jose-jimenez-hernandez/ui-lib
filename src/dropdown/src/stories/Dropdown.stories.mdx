import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Dropdown |

To trigger an operation.

## When To Use



## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 

It accepts all props which native support.
