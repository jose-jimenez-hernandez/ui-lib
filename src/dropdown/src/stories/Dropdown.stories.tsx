import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import { Story } from '@storybook/react';
import Dropdown from '../Dropdown';
import DropdownDoc from './Dropdown.stories.mdx';
import DropdownItem from '../DropdownItem';

const items = [
  { id: '1', name: 'Wade Cooper' },
  { id: '2', name: 'Arlene Mccoy' },
  { id: '3', name: 'Devon Webb' },
  { id: '4', name: 'Tom Cook' },
  { id: '5', name: 'Tanya Fox' },
  { id: '6', name: 'Hellen Schmidt' },
  { id: '7', name: 'Caroline Schultz' },
  { id: '8', name: 'Mason Heaney' },
  { id: '9', name: 'Claudie Smitham' },
  { id: '10', name: 'Emil Schaefer' },
];

export default {
  title: 'Components/Dropdown',
  component: Dropdown,
  parameters: {
    docs: {
      page: (DropdownDoc as any).parameters?.docs.page,
    },
  },
};

const DropdownTemplate: Story = (args: any) => (
  <div className="flex justify-end">
    <Dropdown {...args}>
      {items &&
        items.map((item) => (
          <DropdownItem key={item.id} label={item.name} separator={true} value={item}></DropdownItem>
        ))}
    </Dropdown>
  </div>
);

export const DefaultDropdown = DropdownTemplate.bind({});
DefaultDropdown.args = {};
