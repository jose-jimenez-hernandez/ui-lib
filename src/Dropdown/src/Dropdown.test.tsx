import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';
import * as stories from './stories/Dropdown.stories';

const { DefaultDropdown } =
  composeStories(stories);

it('renders correctly', () => {
  const state = render(<DefaultDropdown />).baseElement;

  expect(state).toMatchSnapshot();
});
