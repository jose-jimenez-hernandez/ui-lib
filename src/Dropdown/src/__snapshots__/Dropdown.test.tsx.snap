// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <div
      class="flex justify-end"
    >
      <div
        class="relative inline-block text-left"
      >
        <div>
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="inline-flex justify-center w-full rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-sm font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-100 focus:ring-primary-500"
            id="headlessui-menu-button-1"
            type="button"
          >
            <span
              class="sr-only"
            >
              Options
            </span>
            <svg
              aria-hidden="true"
              class="h-5 w-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 6a2 2 0 110-4 2 2 0 010 4zM10 12a2 2 0 110-4 2 2 0 010 4zM10 18a2 2 0 110-4 2 2 0 010 4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
